test/sql/aggregate/aggregates/test_distinct_ungrouped.test

```diff
@@ -0,0 +1,191 @@
+# name: test/sql/aggregate/aggregates/test_distinct_ungrouped.test
+# description: DISTINCT aggregations, without GROUP BY
+# group: [aggregates]
+
+# Since these tests are made to test the ungrouped operator, and not necessarily the functions themselves
+# This test will mostly focus on the order and mixing of distinct and non-distinct aggregates
+# And not on variation between types and functions
+
+#distinct aggregate =	'D'
+#regular aggregate =	'-'
+
+statement ok
+PRAGMA enable_verification
+
+statement ok
+create table tbl as
+	(select i%50 as i, i%100 as j from range(50000) tbl(i))
+;
+
+# D
+query I
+select
+	count(distinct i)
+from tbl;
+----
+50
+
+# D--
+query III
+select
+	sum(distinct i),
+	sum(i),
+	sum(j)
+from tbl;
+----
+1225	1225000	2475000
+
+# --D
+query III
+select
+	sum(i),
+	sum(j),
+	sum(distinct i)
+from tbl;
+----
+1225000	2475000	1225
+
+# -D-
+query III
+select
+	sum(i),
+	sum(distinct i),
+	sum(j)
+from tbl;
+----
+1225000	1225	2475000
+
+# D-D
+query III
+select
+	sum(distinct i),
+	count(j),
+	sum(distinct j)
+from tbl;
+----
+1225	50000	4950
+
+#-D-D
+query IIII
+select
+	sum(j),
+	sum(distinct i),
+	count(j),
+	sum(distinct j)
+from tbl;
+----
+2475000	1225	50000	4950
+
+#-D-D
+query IIII
+select
+	sum(j),
+	sum(distinct i),
+	count(j),
+	sum(distinct j)
+from tbl;
+----
+2475000	1225	50000	4950
+
+#D-D-
+query IIII
+select
+	sum(distinct i),
+	count(j),
+	sum(distinct j),
+	sum(j)
+from tbl;
+----
+1225	50000	4950	2475000
+
+# These next tests will repeat the previous test, with the addition of filters
+# filtered =		'F'
+# not filtered =	'-'
+
+# D
+# F
+query I
+select
+	count(distinct i) FILTER (WHERE i >= 20)
+from tbl;
+----
+30
+
+# D--
+# -FF
+query III
+select
+	sum(distinct i),
+	sum(i) FILTER (WHERE j < 20),
+	sum(j) FILTER (WHERE i >= 20)
+from tbl;
+----
+1225	95000	1785000
+
+# --D
+# -FF
+query III
+select
+	sum(i),
+	sum(j) FILTER (WHERE j == 0),
+	sum(distinct i) FILTER (WHERE i == 0)
+from tbl;
+----
+1225000	0	0
+
+# -D-
+# F-F
+query III
+select
+	sum(i) FILTER (WHERE j == 5),
+	sum(distinct i),
+	sum(j) FILTER (WHERE i == 5)
+from tbl;
+----
+2500	1225	30000
+
+# D-D
+# F-F
+query III
+select
+	sum(distinct i) FILTER (WHERE i == 5),
+	count(j),
+	sum(distinct j) FILTER (WHERE i == 5)
+from tbl;
+----
+5	50000	60
+
+#-D-D
+#FF--
+query IIII
+select
+	sum(j) FILTER (WHERE j == 5),
+	sum(distinct i) FILTER (WHERE j == 5),
+	count(j),
+	sum(distinct j)
+from tbl;
+----
+2500	5	50000	4950
+
+#-D-D
+#F--F
+query IIII
+select
+	sum(j) FILTER (WHERE i == 5),
+	sum(distinct i),
+	count(j),
+	sum(distinct j) FILTER (WHERE j == 5)
+from tbl;
+----
+30000	1225	50000	5
+
+#D-D-
+query IIII
+select
```

Comment:
Could we add a test where we perform distinct aggregates on an expression, e.g.:

```sql
CREATE TABLE tbl AS SELECT * FROM range(1000000) tbl(i);
SELECT COUNT(distinct i % 5) from tbl;
```

As well as on a correlated subquery:

```sql
CREATE TABLE tbl AS SELECT * FROM range(1000000) tbl(i);
SELECT COUNT(distinct (SELECT i%5)) from tbl;
```

And a `DISTINCT` with an aggregate with multiple parameters:

```sql
CREATE TABLE tbl AS SELECT * FROM range(1000000) tbl(i);
SELECT ARG_MIN(distinct i, i%5) from tbl;
```